Examples/MAX32672/WearLeveling/main.c

```diff
@@ -42,82 +42,77 @@
-#define FLASH_STORAGE_START_PAGE 8 ///< Internal storage first flash memory block
-#define FLASH_STORAGE_PAGE_CNT 8 ///< Flash memory blocks reserved for the internal storage
-#define FLASH_STORAGE_START_ADDR \
-    MXC_FLASH_PAGE_ADDR(FLASH_STORAGE_START_PAGE) ///< Internal storage start address
-#define FLASH_STORAGE_SIZE FLASH_STORAGE_PAGE_CNT *MXC_FLASH_PAGE_SIZE ///< Internal storage size
-
-// When set to 1 performs full storage erase and test data write
-#ifndef FULL_WRITE_TEST
-#define FULL_WRITE_TEST 0
-#endif
+// Modifiable LittleFS macros
+#define LFS_PGE_CNT 8 ///< Number of flash pages reserved for LittleFS (Valid values 1-112)
 
-// When set to 1 performs full storage test data read. Passes only if FULL_WRITE_TEST was performed before
-#ifndef FULL_READ_TEST
-#define FULL_READ_TEST 0
-#endif
+// Non-modifiable LittleFS macros
+#define LFS_START_PGE 16 ///< Page number of the first LittleFS block
+#define LFS_START_ADDR MXC_FLASH_PAGE_ADDR(LFS_START_PGE) ///< Start address of LittleFS
+#define LFS_SIZE (LFS_PGE_CNT * MXC_FLASH_PAGE_SIZE) ///< Size of LittleFS
+
+// Set to 1 to perform erase, test data write, and test read back/verify over LittleFS Flash space
+#define FLASH_TEST 0
+
+// Size of FLASH_TEST data array
+#define TESTSIZE (MXC_FLASH_PAGE_SIZE * LFS_PGE_CNT / 4) ///< LFS_PGE_CNT pages of 32 bit samples
 
 /***** Globals *****/
 uint32_t testdata[TESTSIZE]; ///< Test data buffer
 
 // variables used by the filesystem
 lfs_t lfs; ///< File system instance
-uint32_t start_block =
-    FLASH_STORAGE_START_PAGE; ///< Internal memory start block to be passed to flash functions by littlefs
-// configuration of the filesystem is provided by this struct
+uint32_t start_block = LFS_START_PGE; ///< Internal memory start block
+
+// configuration of the filesystem
 const struct lfs_config cfg = {
     .context = &start_block,
-    // block device operations
+
+    // Function pointers to flash operations (in flash.h/.c)
     .read = flash_read,
     .prog = flash_write,
     .erase = flash_erase,
     .sync = flash_sync,
 
     // block device configuration
-    .read_size = 1,
-    .prog_size = 16,
-    .block_size = MXC_FLASH_PAGE_SIZE,
-    .block_count = FLASH_STORAGE_PAGE_CNT,
+    .read_size = 1, //< 1-byte Flash reads
+    .prog_size = 16, //< 16-byte Flash writes
+    .block_size = MXC_FLASH_PAGE_SIZE, //< LFS block = 1 Flash page
+    .block_count = LFS_PGE_CNT, //< Num flash pages reserved for LFS
     .cache_size = 16,
     .lookahead_size = 16,
     .block_cycles = 500,
 };
 
 //******************************************************************************
-/**
- * @brief Application entry point
- * @return Exit code
- */
 int main(void)
 {
     int error_status = E_NO_ERROR;
 
+    // Disable ECC on Flash 0 if necessary
+    uint32_t ecc_status = (MXC_TRIMSIR->bb_sir2 & MXC_F_TRIMSIR_BB_SIR2_FL0ECCEN);
+    if (ecc_status) {
+        MXC_TRIMSIR->bb_sir2 &= ~MXC_F_TRIMSIR_BB_SIR2_FL0ECCEN;
+    }
+
```

Comment:
I'm surprised we don't have a library for the ECC (Status/Enable/Disable). Noted as a future task.